app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -444,6 +444,12 @@ def public_key=(key)
       attrs[:public_key] = key
     end
 
+    def validate_vm_attrs(vm_attrs)
+      vm_attrs['cluster'] = get_ovirt_id(clusters, vm_attrs['cluster'])
+      vm_attrs['template'] = get_ovirt_id(templates, vm_attrs['template'])
+      vm_attrs
+    end
```

Comment:
Shouldn't this be done as part of `normalize_vm_attrs`?

Reply:
I wanted to create a separation between the validation of the VM attributes to the normalize_vm_attrs method because I wanted to add it before saving.
Do you think I should add it in normalize_vm_attrs and put this method before save?

Reply:
This doesn't look like a validation. Validations don't throw an exception, they add an error message when some part of the input is incorrect, and run in the `validate` callback stage.  (see https://guides.rubyonrails.org/active_record_validations.html#custom-methods)
This functions appears to get the ovirt id for the cluster and template, and runs in `before_save`, which is executed after validations (see https://guides.rubyonrails.org/active_record_callbacks.html#available-callbacks). Also, looks like `get_ovirt_id` throws a `Foreman::Exception` which while aborting the save, does not show up as a validation error pointing to correct field, but rather as a server error.
If you want this to be a validation, you run it as a validation in the compute attribute model, catch the exception and modify the error object accordingly.

Reply:
Ok, I will fix it.